src/world/area_hos/hos_05/A2AA70.c

```diff
@@ -1,3 +1,10 @@
 #include "hos_05.h"
 
-INCLUDE_ASM(s32, "world/area_hos/hos_05/A2AA70", func_80240830_A2AA70);
+ApiStatus func_80240830_A2AA70(Evt* script, s32 isInitialCall) {
+    gPlayerData.starBeamLevel = 1;
+    gPlayerData.curHP = gPlayerData.curMaxHP;
+    gPlayerData.curFP = gPlayerData.curMaxFP;
+    gPlayerData.specialBarsFilled = gPlayerData.maxStarPower * 256;
```

Comment:
we should probably keep track of this conversion factor at some point, defining POW_POINTS_PER_BAR = 256?